### ISSUE TITLE
Plots generated in a loop

### ISSUE THREAD

**Issue by dpo** [tbreloff/Plots.jl#418]
Sorry if this is obvious but the following doesn't produce any plot on my machine (Julia 0.4.5, OSX 10.9):
```julia
using Plots

function plot_stuff()
  fig = plot(reuse=false)
  plot!(collect(1:10), rand(10), color=:blue, label="blue")
  plot!(collect(1:10), rand(10), color=:red, label="red")
  title!("Random Stuff")
  return fig
end

for i = 1 : 4
  fig = plot_stuff()
end
```
Is that to be expected? If I have a single call to `plot_stuff()` instead of a loop, a window appears.

**Comment by KristofferC** [comment]
what happens if you put `fig`` in the  end so it's `show` method is executed?

**Comment by dpo** [comment]
Do you mean after the loop? We hit one of Julia's mysteries:
```
julia> fig
ERROR: UndefVarError: fig not defined
```
Variables created inside a loop are somehow discarded when the loop ends. Perhaps that's the problem, but I haven't managed to create a variable `fig` before the loop so the plots show up.

**Comment by KristofferC** [comment]
Put `local fig` before the loop.

**Comment by dpo** [comment]
Seems we can't do that outside of a function:
```
ERROR: LoadError: syntax: misplaced "local" declaration
```

**Comment by tbreloff** [comment]
Please read the manual: https://juliaplots.github.io/output/

If you did this:

```julia
for i=1:10
    x = 1
end
```

would you be surprised when nothing was output to the screen?  This is no different.  The plot is displayed when it is "shown", either explicitly or when returned to the REPL prompt.

**Comment by tbreloff** [comment]
And just in case you didn't realize, your method could be much more compact:

```julia
function plot_stuff()
    plot(rand(10,2), color=[:blue :red], label=["blue" "red"], 
                     title="Random Stuff", reuse=false)
end
```

**Comment by dpo** [comment]
Thanks. Sorry for the bother. Seems I missed that part of the manual.

> If you did this: [...] would you be surprised when nothing was output to the screen?

Well, yes and no. I guess I was expecting plots to show up unless told not to. But I now realize it's the other way around. Fair enough.

> And just in case you didn't realize, your method could be much more compact:

Thanks. That was just a simplified version of my actual script, which needs to construct plots one curve at a time.

**Comment by superkerokero** [comment]
It seems that when I try to plot it using Jupyter Notebook, like the following code:
`
using Plots
pyplot()
for i=1:100
    plot(rand(10))
end
`
Nothing shows up either(inside the notebook). If I add a 'gui()' inseide the loop, only the last plot is show... in a standalone window.
Looks like a bug to me.

**Comment by essenciary** [comment]
@tbreloff Now the link https://juliaplots.github.io/output/ ends up in a 404 and I'm having the same problem. It would've helped greatly to mention how to get the plot to show, for the benefit of posterity and yours truly.

**Comment by essenciary** [comment]
OK, found it. Here's for posterity: 

> A Plot is only displayed when returned (a semicolon will suppress the return), or if explicitly displayed with `display(plt)`, `gui()`, or by adding `show = true` to your plot command.

**Comment by rulatir** [comment]
Jesus Christ, can someone just write the exact incantation? Please take the original example and modify it minimally so that after modification it will DoTheRightThing™, the right thing being defined by the original reporter's clear desire to Have Things Actually Be Shown.

**Comment by mkborregaard** [comment]
For even more posterity, here's the correct syntax to plot in a for loop
```julia

p = plot(rand(10))
for i in 1:4
    plot!(p, rand(10))
end
p #displays the plot
```

**Comment by Amend-1634** [comment]
'return current()' within the function may do the work! You may give it a try~
> using Plots
> function plot_stuff()
>   fig = plot(reuse=false)
>   plot!(collect(1:10), rand(10), color=:blue, label="blue")
>   plot!(collect(1:10), rand(10), color=:red, label="red")
>   title!("Random Stuff")
>   return current() 
> end
> 
> for i = 1 : 4
>   fig = plot_stuff()
> end

**Comment by mkborregaard** [comment]
The correct syntax has already been posted, and this issue has been closed for 3.5 years
